src/CommunityPatch/Patches/Perks/Endurance/Riding/TramplerPatch.cs

```diff
@@ -0,0 +1,134 @@
+        && instructions[start + 2].opcode == OpCodes.Stloc_S
+        && instructions[start + 3].opcode == OpCodes.Ldloc_S
+        && instructions[start + 4].opcode == OpCodes.Brfalse_S;
+
+    private static BasicCharacterObject UpdateCorrectCharacterForHorseChargeDamage(Agent agent, ref AttackCollisionData acd, BasicCharacterObject character)
+      => acd.IsHorseCharge && agent.RiderAgent != null ? agent.RiderAgent.Character : character;
+    
+    private static readonly MethodInfo CorrectCharacterForHorseChargeDamageMethod = typeof(TramplerPatch)
+      .GetMethod("UpdateCorrectCharacterForHorseChargeDamage", BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.DeclaredOnly);
+    
+    private static readonly List<LocalVariableInfo> characterLocalVariableInfos = TargetMethodInfo
+      .GetMethodBody().LocalVariables.Where(var => var.LocalType == typeof(BasicCharacterObject)).ToList();
+    
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions) {
+
+      if (characterLocalVariableInfos.Count != 2) {
+        CommunityPatchSubModule.Error($"{typeof(TramplerPatch).Name}: Expected two BasicCharacterObject  local variable instances in the original method."
+          + $" The original code has been changed requiring an update of this patch." + Environment.NewLine);
+        return instructions;
+      }
+      
+      var codes = instructions.ToList();
+      for (var index = 0; index<codes.Count - 5; index++)
+      {
+        if (IsAttackerAgentAssignedAndAgentRiderNotNull(codes, index)) {
+          var attackerCharacterLocalVariableIndex = characterLocalVariableInfos[1].LocalIndex;
+          codes.InsertRange(index+5, new List<CodeInstruction> {
+            new CodeInstruction(OpCodes.Ldarg_1), // Load attackerAgent argument
+            new CodeInstruction(OpCodes.Ldarg_S, 5), // Load attackCollisionData argument
+            new CodeInstruction(OpCodes.Ldloc_S, attackerCharacterLocalVariableIndex), // Load attackerCharacter local variable
+            new CodeInstruction(OpCodes.Call, CorrectCharacterForHorseChargeDamageMethod), // Update attackerCharacter if the collision is a horse charge
+            new CodeInstruction(OpCodes.Stloc_S, attackerCharacterLocalVariableIndex) // Store value in attackerCharacter
+          });
+          return codes.AsEnumerable();
+        }
+      }
+      
+      CommunityPatchSubModule.Error($"{typeof(TramplerPatch).Name}: Could not find the starting point to add new instructions in {TargetMethodInfo}." + Environment.NewLine);
+      return codes.AsEnumerable();
+    }
+    
+    private static bool HeroHasPerk(BasicCharacterObject character, PerkObject perk)
+      => (character as CharacterObject)?.GetPerkValue(perk) ?? false;
+
+    private static int TramplerDamageModifier(int baseDamage) => baseDamage + baseDamage/2; 
```

Comment:
```suggestion
    private static int TramplerDamageModifier(int baseDamage) => (int) MathF.Round(baseDamage * 1.5f);
```
Is this the same except with rounding?

Reply:
Yeah it is the same, although rounding is not necessary in this context since we will always have a bonus integer of X or X + 0.5 which in an integer division will be X. But, if the bonus is bound to change, then rounding might prevent us an calculation error.

Reply:
Alright, let's do this as you mentionned.